text/0043-block-metadata.md

```diff
@@ -0,0 +1,238 @@
+    /// Metadata signed by the consensus node.
+    BlockMetadataContents contents = 1;
+
+    /// Message signing key.
+    Ed25519Public node_key = 2;
+
+    /// Signature using `node_key` over the Digestible encoding of `contents`.
+    Ed25519Signature signature = 3;
+}
+
+// NB: Only addition is field #4.
+message ArchiveBlockV1 {
+    /// Block.
+    Block block = 1;
+
+    /// Contents of the block.
+    BlockContents block_contents = 2;
+
+    /// Block signature.
+    BlockSignature signature = 3;
+
+    /// Additional signed metadata about this block.
+    BlockMetadata metadata = 4;
+}
+```
+
+Since this is a protobuf-additive change, we don't need a new `ArchiveBlock`
+version/type.
+
+## Signing metadata
+The metadata signature will always differ from the block signature. The block
+signature is generated by the enclave using a private key that never leaves the
+enclave, while the metadata is signed by the "untrusted" code outside the
+enclave using the SCP message signing key, since the enclave does not have this
+metadata.
+
+The metadata signature is generated from the `Digestible` encoding of
+`BlockMetadataContents`, since the protobuf wire format is not a canonical
+encoding.
+
+## Signing key history
+MobileCoin Foundation will publish a mapping/list of node_id to a range of block
+indexes, which node operators and consumers can use to verify the metadata
+signatures.
+
+This mapping will contain multiple node_id:block_range entries
+
+#### Key History TOML format
+```toml
+[[node]]
+# Pub key, as hex-encoded or PEM-encoded data or a PEM file path.
+message_signing_pub_key = "..."
+# Index of the first block signed with this key.
+first_block_index = X
+# Optional index of the last block signed with this key.
+last_block_index = Y
+```
+
+#### Key History JSON format
+```json
+{
+  "node": [
+    {
+      "message_signing_pub_key": "...",
+      "first_block_index": 0,
+      "last_block_index": 42
+    }
+  ]
+}
+```
+
+## Historical AVRs
+With `BlockMetadata`, consumers of the blockchain may not need a Watcher DB.
+Ledger DB can default to reading AVRs from `BlockMetadata` (using the
+aforementioned signing key history to authenticate the metadata signature),
+using the AVR to verify the enclave key used for the block signature, and fall back
+to a lookup table containing AVRs for older blocks.
+
+MobileCoin Foundation will publish a bootstrap file defining this lookup table
+with historical consensus enclave AVRs, up to when consensus nodes publish
+`BlockMetadata` with their AVRs.
+
+#### AVR History TOML format
+```toml
+[[node]]
+# Responder ID for the consensus node.
+responder_id = ""
+# Index of the first block signed with this enclave.
+first_block_index = X
+# Optional index of the last block signed with this enclave.
+last_block_index = Y
+[node.avr]
+# Hex-encoded signature bytes
+sig = "..."
+# An array of hex-encoded pub key bytes
+chain = ["x", "y"]
+# IAS HTTP response body
+http_body = '{...}'
+```
```

Comment:
Is there an advantage to doing this vs. `avr_sig avr_chain`, `avr_http_body`?

Reply:
It maintains the grouping of related fields by type rather than a convention of common prefix, and keeps the JSON and TOML representations compatible.

With that said, I'm open to changing to `avr_*` if that proves better.